source/global-this.d.ts

```diff
@@ -0,0 +1,40 @@
+/**
+Aids in defining global symbols in a narrow scope. This type is compatible with casting `window`, `globalThis`, and `self`.
+
+@example
+```
+import type {GlobalThis} from 'type-fest';
+
+interface ExtraGlobals extends GlobalThis {
+	readonly GLOBAL_TOKEN: string;
+}
+
+(window as ExtraGlobals).GLOBAL_TOKEN;
+// or
+(globalThis as ExtraGlobals).GLOBAL_TOKEN;
+// or
+(self as ExtraGlobals).GLOBAL_TOKEN;
+```
+
+@category Type
+*/
+export type GlobalThis = (typeof globalThis) & WindowSansInfinityNorNaN;
+
+/**
+Changes the type of `Infinity` and `NaN` to `never` on `Window`.
+
+Avoids the error "Property 'Infinity' of type 'number' is not assignable to numeric index type 'Window'."
+*/
+// This \@ts-ignore is present as the "dom" lib is not guarenteed to be loaded
+// and we don't want to globaly declare Window for libs that don't need it.
+// eslint-disable-next-line @typescript-eslint/prefer-ts-expect-error
+// @ts-ignore
+type WindowSansInfinityNorNaN = ExcludeAny<Window & {
+	Infinity: never;
+	NaN: never;
+}>;
+
+/**
+If `T` is `any`, returns unknown. Safe as `SomeType & unknown` is `SomeType`.
+*/
+type ExcludeAny<Type> = Type extends never ? unknown : Type;
```

Comment:
Is it expected that it returns `unknown` and not `never` when the input is `any`?

Alternative name: `AnyToUnknown`

Reply:
This should be moved to internal.d.ts

Reply:
Maybe: `type AnyToUnknown<T> = unknown extends T ? unknown : T;`

Reply:
No longer relevant as `ExcludeAny` is no longer required.